Archive/Imo/Imo1987Q1.lean

```diff
@@ -46,8 +46,15 @@ def fixedPointsEquiv : { σx : α × Perm α // σx.2 σx.1 = σx.1 } ≃ Σ x :
 
 theorem card_fixed_points :
     card { σx : α × Perm α // σx.2 σx.1 = σx.1 } = card α * (card α - 1)! := by
-  simp [card_congr (fixedPointsEquiv α), card_perm, Finset.filter_not, Finset.card_sdiff,
-    Finset.filter_eq', Finset.card_univ]
+  simp only [card_congr (fixedPointsEquiv α), card_sigma, card_perm]
```

Comment:
(Minor comment, but when leaving comments in huge PRs, it helps if you click and drag through all the changed lines before commenting. This means that the context is visible on the "conversation page", rather than having to wade through the diffs to find the context.)

Reply:
I'm sorry. I didn't know there was that functionality. Thanks.

Reply:
I think we can live with this change, and probably we will add eta-reduced lemmas about membership over time. I'm not proposing to do this as part of this PR, but if we add a lemma
```lean
theorem Set.mem_compl_eq {α : Type*} {s : Set α} : (· ∈ sᶜ) = (· ∉ s) := rfl
```
then the proof of `card_fixed_points` is just
```lean
  simp [Set.mem_compl_eq, card_congr (fixedPointsEquiv α), card_perm, Finset.filter_ne']
```

Reply:
That's a good catch.